lib/STM_domain.mli

```diff
@@ -17,18 +22,47 @@ module Make : functor (Spec : STM.Spec) ->
         The three {!Spec.cmd} components are generated with [arb0], [arb1], and [arb2], respectively.
         Each of these take the model state as a parameter. *)
 
+    val arb_triple_asym : int -> int -> (Spec.state -> Spec.cmd QCheck.arbitrary) -> (Spec.state -> Spec.cmd QCheck.arbitrary) -> (Spec.state -> Spec.cmd QCheck.arbitrary) -> (Spec.cmd list * Spec.cmd list * Spec.cmd list) QCheck.arbitrary
+    (** [arb_triple_asym seq_len par_len arb0 arb1 arb2] creates a triple [cmd]
+        generator like {!arb_triple}. It differs in that the resulting printer
+        is asymmetric, printing [arb1]'s result below [arb0]'s result and
+        printing [arb2]'s result to the right of [arb1]'s result. *)
+
     val interp_sut_res : Spec.sut -> Spec.cmd list -> (Spec.cmd * STM.res) list
     (** [interp_sut_res sut cs] interprets the commands [cs] over the system {!Spec.sut}
         and returns the list of corresponding {!Spec.cmd} and result pairs. *)
 
     val agree_prop_par : Spec.cmd list * Spec.cmd list * Spec.cmd list -> bool
-    (** Parallel agreement property based on {!Stdlib.Domain} *)
+    (** Parallel agreement property based on {!Stdlib.Domain}.
+        [agree_prop_par (seq_pref, tl1, tl2)] first interprets [seq_pref]
+        and then spawns two parallel, symmetric domains interpreting [tl1] and
+        [tl2] simultaneously.
+
+        @return [true] if there exists a sequential interleaving of the results
+        which agrees with a model interpretation. *)
+
+    val agree_prop_par_asym : Spec.cmd list * Spec.cmd list * Spec.cmd list -> bool
+    (** Asymmetric parallel agreement property based on {!Stdlib.Domain}.
+        [agree_prop_par_asym (seq_pref, tl1, tl2)] first interprets [seq_pref],
+        and then interprets [tl1] while a spawned domain interprets [tl2]
+        in parallel with the parent domain.
+
+        @return [true] if there exists a sequential interleaving of the results
+        which agrees with a model interpretation. *)
 
     val agree_test_par : count:int -> name:string -> QCheck.Test.t
     (** Parallel agreement test based on {!Stdlib.Domain} which combines [repeat] and [~retries] *)
 
     val neg_agree_test_par : count:int -> name:string -> QCheck.Test.t
-    (** A negative agreement test (for convenience). Accepts two labeled parameters:
+    (** A negative parallel agreement test (for convenience). Accepts two labeled parameters:
         [count] is the test count and [name] is the printed test name. *)
 
+    val agree_test_par_asym : count:int -> name:string -> QCheck.Test.t
+    (** Asymmetric parallel agreement test based on {!Stdlib.Domain} and {!agree_prop_par_sym}
+        which combines [repeat] and [~retries] *)
+
+    val neg_agree_test_par_asym : count:int -> name:string -> QCheck.Test.t
+    (** A negative asymmetric parallel agreement test (for convenience).
+        Accepts two labeled parameters:
```

Comment:
I just notice this line because it appears in the diff: the 4 *`agree_test_par`* accept those labelled parameters but it is documented on two of them. The labels are self-explanatory, though.

Reply:
Fair enough! Addressed in b6a471a